File: Data Structures & Algorithms/jump-game-ii/submission-0.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int n = nums.size();
        int currTarget = n - 1;
        int tmpTarget = n - 1;
        int result = 0;


        while (currTarget != 0) {
            for (int i = currTarget; i >=0; --i) {
                if (nums[i] + i >= currTarget) {
                    tmpTarget = i;
                }
            }
            currTarget = tmpTarget;
            ++result;
        }

        return result;
    }
};

File: Data Structures & Algorithms/subsets/submission-0.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        result = [[]]

        for num in nums:
            if [num] not in nums:
                result.append([num])

                for s in result:
                    new_set = list(set([num] + s))
                    contains = False
                    for s in result:
                        if set(s) == set(new_set):
                            contains = True
                    
                    if not contains:
                        result.append(new_set)

        return result
            
        